editor/src/components/canvas/dom-lookup.ts

```diff
@@ -121,12 +131,22 @@ export function findFirstParentWithValidElementPathInner(
       for (const validPath of validStaticElementPaths) {
```

Comment:
I _think_ I just realized something! the reason this function needs to be called with both `allow-descendants-from-path` and `search-dom-only` is because the `allow-descendants-from-path` case cannot return with the dom element! If that is true, then I think this code could be vastly simplified here, for future ease of maintenance